pppoe/pppoe.go

```diff
@@ -102,13 +119,13 @@ func New(ctx context.Context, ifName string) (*Conn, error) {
 // Conns don't have an interesting local address to share, so this
 // returns nil for now.
 func (c *Conn) LocalAddr() net.Addr {
-	return nil
+	return &c.localAddr
```

Comment:
we try to keep variables in the types that implement the methods/interface rather than change it last second like this.

for instance, instead of code like:

```go
         t := T{
               ....
         }
         return &t, nil
```

We write:

```go
         t := &T{
               ....
         }
         return t, nil
```

It prevents people from accidentally doing copies by value meanwhile, and is less to think about for readers.

So unless it really matters for performance, I'd make the two Addrs be `*Addr` in the struct, or make the Addr value type implement the interface. The former is probably more efficient.